src/libraries/Native/Unix/System.Globalization.Native/pal_casing.c

```diff
@@ -150,6 +150,24 @@ void GlobalizationNative_ChangeCaseTurkish(
     }
 }
 
+void GlobalizationNative_InitOrdinalCasingPage(int32_t pageNumber, UChar* pTarget)
+{
+    pageNumber <<= 8;
+    for (int i = 0; i < 256; i++)
+    {
+        // Unfortunately, to ensure one-to-one simple mapping we have to call u_toupper on every character.
+        // Using string casing ICU APIs cannot give such results even when using NULL locale to force root behavior.
```

Comment:
FWIW, this is because Unicode itself doesn't have 1:1 case mapping.

Reply:
Actually if limit the functionality to UnicodeData.txt, it will be 1:1. Yes, I understand in general Unicode casing is not 1:1.